frontend/types/api.ts

```diff
@@ -0,0 +1,58 @@
+// Code generated by tygo. DO NOT EDIT.
+import * as node from "./node"
+
+//////////
+// source: node.go
+
+export interface MigrateRequest {
+  target_version: string;
+}
+export interface GetNodeResponse {
+  state: node.State;
+}
+export interface PostAddUserRequest {
+  user_id: string;
+  handle: string;
+  password: string;
+  email: string;
+  certificate: string;
+}
+export interface PostAddUserResponse {
+  pds_create_account_response: { [key: string]: any};
+}
+export interface PostAppRequest {
+  app_installation?: node.AppInstallation;
+  reverse_proxy_rules: (node.ReverseProxyRule | undefined)[];
+}
+export interface PostProcessRequest {
+  app_id: string;
+}
+
+//////////
+// source: pds.go
+
+export interface PDSInviteCodeResponse {
+  code: string;
+}
+export interface PDSCreateAccountRequest {
+  email: string;
+  handle: string;
+  password: string;
+  inviteCode: string;
+}
+export type PDSCreateAccountResponse = { [key: string]: any};
+export interface PDSCreateSessionRequest {
+  identifier: string; // email or handle
+  password: string;
+}
+export type PDSCreateSessionResponse = { [key: string]: any};
+export interface PDSGetRecordRequest {
```

Comment:
where do these get generated from? from Go territory?

Reply:
Yup, checkout the changes to the Makefile to see the rules that generate these.